pkg/resources/connection.go

```diff
@@ -0,0 +1,321 @@
+		},
+	}
+}
+
+func CreateContextConnection(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+
+	id, err := sdk.ParseAccountObjectIdentifier(d.Get("name").(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	request := sdk.NewCreateConnectionRequest(id)
+
+	if v, ok := d.GetOk("as_replica_of"); ok {
+		if externalObjectId, err := sdk.ParseExternalObjectIdentifier(v.(string)); err != nil {
+			request.WithAsReplicaOf(externalObjectId)
+		} else {
+			return diag.FromErr(err)
+		}
+	}
+
+	if v, ok := d.GetOk("comment"); ok {
+		request.WithComment(v.(string))
+	}
+
+	err = client.Connections.Create(ctx, request)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(helpers.EncodeResourceIdentifier(id))
+
+	if v, ok := d.GetOk("is_primary"); ok {
+		err := client.Connections.Alter(ctx, sdk.NewAlterConnectionRequest(id).
+			WithPrimary(v.(bool)),
+		)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	if v, ok := d.GetOk("enable_failover_to_accounts"); ok {
+		enableFailoverConfig := v.([]any)
+
+		enableFailoverToAccountsList := make([]sdk.AccountIdentifier, 0)
+		for _, enableToAccount := range enableFailoverConfig {
+			accountInConfig := enableToAccount.(string)
+			accountIdentifier := sdk.NewAccountIdentifierFromFullyQualifiedName(accountInConfig)
+
+			enableFailoverToAccountsList = append(enableFailoverToAccountsList, accountIdentifier)
+		}
+
+		err := client.Connections.Alter(ctx, sdk.NewAlterConnectionRequest(id).
+			WithEnableConnectionFailover(*sdk.NewEnableConnectionFailoverRequest().
+				WithToAccounts(enableFailoverToAccountsList)),
+		)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	return ReadContextConnection(false)(ctx, d, meta)
+}
+
+func ReadContextConnection(withExternalChangesMarking bool) schema.ReadContextFunc {
+	return func(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+		client := meta.(*provider.Context).Client
+		id, err := sdk.ParseAccountObjectIdentifier(d.Id())
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		connection, err := client.Connections.ShowByID(ctx, id)
+		if err != nil {
+			if errors.Is(err, sdk.ErrObjectNotFound) {
+				d.SetId("")
+				return diag.Diagnostics{
+					diag.Diagnostic{
+						Severity: diag.Warning,
+						Summary:  "Failed to retrieve connection. Target object not found. Marking the resource as removed.",
+						Detail:   fmt.Sprintf("Connection name: %s, Err: %s", id.FullyQualifiedName(), err),
+					},
+				}
+			}
+			return diag.Diagnostics{
+				diag.Diagnostic{
+					Severity: diag.Error,
+					Summary:  "Failed to retrieve connection.",
+					Detail:   fmt.Sprintf("Connection name: %s, Err: %s", id.FullyQualifiedName(), err),
+				},
+			}
+		}
+
+		if withExternalChangesMarking {
+			if err := handleExternalChangesToObjectInShow(d,
+				outputMapping{"is_primary", "is_primary", connection.IsPrimary, connection.IsPrimary, nil},
```

Comment:
Not sure if this is needed

Reply:
it could but there is a disadvantage to this approach, meaning `is_primary` would have to be required.
For now it is optional and provided only with `as_replica_of`. So when we try to set it with `d.Set("is_primary", connection.IsPrimary)` TF will perform action changing `is_primary = null` to `is_primary = true`

@sfc-gh-jmichalak mentioned [here](https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/3162#discussion_r1822189178) that maybe it should be provided to both replicated and primary connection? wdyt?

Reply:
I think you are already handling it correctly, so this can be resolved.